```
       last 25 log lines:
       > CMake Error at /nix/store/p9w37j80q80qid636hpgdwpknmrya75h-cmake-3.31.5/share/cmake-3.31/Modules/FetchContent.cmake:1906 (message):
       >   CMake step for protobuf failed: 1
       > Call Stack (most recent call first):
       >   /nix/store/p9w37j80q80qid636hpgdwpknmrya75h-cmake-3.31.5/share/cmake-3.31/Modules/FetchContent.cmake:1609 (__FetchContent_populateSubbuild)
       >   /nix/store/p9w37j80q80qid636hpgdwpknmrya75h-cmake-3.31.5/share/cmake-3.31/Modules/FetchContent.cmake:2145:EVAL:2 (__FetchContent_doPopulation)
       >   /nix/store/p9w37j80q80qid636hpgdwpknmrya75h-cmake-3.31.5/share/cmake-3.31/Modules/FetchContent.cmake:2145 (cmake_language)
       >   /nix/store/p9w37j80q80qid636hpgdwpknmrya75h-cmake-3.31.5/share/cmake-3.31/Modules/FetchContent.cmake:2384 (__FetchContent_Populate)
       >   luau/fuzz/CMakeLists.txt
       >   luau/fuzz/CMakeLists.txt:68 (find_package)
       >
       >
       > CMake Error at luau/fuzz/CMakeLists.txt:68 (find_package):
       >   Could not find a package configuration file provided by "Protobuf" with any
       >   of the following names:
       >
       >     ProtobufConfig.cmake
       >     protobuf-config.cmake
       >
       >   Add the installation prefix of "Protobuf" to CMAKE_PREFIX_PATH or set
       >   "Protobuf_DIR" to a directory containing one of the above files.  If
       >   "Protobuf" provides a separate development package or SDK, be sure it has
       >   been installed.
       >
       >
       > -- Configuring incomplete, errors occurred!
       For full logs, run:
         nix log /nix/store/w9ryd8jgnzkc3l73mavavgrw6jzkih4w-luau-lsp-1.45.0.drv
error: builder for '/nix/store/w9ryd8jgnzkc3l73mavavgrw6jzkih4w-luau-lsp-1.45.0.drv' failed with exit code 1
[0 built (2 failed), 0.0 MiB DL]
```

--- a/maintainers/maintainer-list.nix
+++ b/maintainers/maintainer-list.nix
@@ -1689,6 +1689,11 @@
     githubId = 81317317;
     name = "Anna Aurora";
   };
+  anninzy = {
+    github = "anninzy";
+    githubId = 143312793;
+    name = "Annin";
+  };
   anoa = {
     matrix = "@andrewm:amorgan.xyz";
     email = "andrew@amorgan.xyz";

--- a/pkgs/applications/editors/vim/plugins/overrides.nix
+++ b/pkgs/applications/editors/vim/plugins/overrides.nix
@@ -118,6 +118,8 @@
   # typst-preview dependencies
   tinymist,
   websocat,
+  # luau-lsp-nvim dependencies
+  luau-lsp,
 }:
 self: super:
 let
@@ -1690,9 +1692,7 @@ in
 
   luau-lsp-nvim = super.luau-lsp-nvim.overrideAttrs {
     dependencies = [ self.plenary-nvim ];
-
-    # TODO: add luau-lsp to nixpkgs (#395892)
-    # runtimeDeps = [ luau-lsp ];
+    runtimeDeps = [ luau-lsp ];
   };
 
   magma-nvim = super.magma-nvim.overrideAttrs {

--- a/pkgs/by-name/lu/luau-lsp/package.nix
+++ b/pkgs/by-name/lu/luau-lsp/package.nix
@@ -1,0 +1,70 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  cmake,
+  versionCheckHook,
+  nix-update-script,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "luau-lsp";
+  version = "1.45.0";
+
+  src = fetchFromGitHub {
+    owner = "JohnnyMorganz";
+    repo = "luau-lsp";
+    tag = finalAttrs.version;
+    hash = "sha256-OJAjTy0vTRb43TTiPeXafWq4kjIpnDXoTprVzbMnaWQ=";
+    fetchSubmodules = true;
+  };
+
+  NIX_CFLAGS_COMPILE = "-Wno-error";
+
+  cmakeFlags = lib.optionals stdenv.hostPlatform.isDarwin [
+    (lib.cmakeFeature "CMAKE_OSX_ARCHITECTURES" stdenv.hostPlatform.darwinArch)
+  ];
+
+  nativeBuildInputs = [ cmake ];
+
+  buildPhase = ''
+    runHook preBuild
+
+    cmake --build . --target Luau.LanguageServer.CLI --config Release
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    install -D luau-lsp $out/bin/luau-lsp
+
+    runHook postInstall
+  '';
+
+  nativeInstallCheckInputs = [ versionCheckHook ];
+  versionCheckProgramArg = "--version";
+  doInstallCheck = true;
+
+  passthru.updateScript = nix-update-script { };
+
+  meta = {
+    description = "Language Server Implementation for Luau";
+    homepage = "https://github.com/JohnnyMorganz/luau-lsp";
+    downloadPage = "https://github.com/JohnnyMorganz/luau-lsp/releases/tag/${finalAttrs.version}";
+    changelog = "https://github.com/JohnnyMorganz/luau-lsp/blob/${finalAttrs.version}/CHANGELOG.md";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [
+      anninzy
+      HeitorAugustoLN
+    ];
+    mainProgram = "luau-lsp";
+    platforms = lib.platforms.all;
+    badPlatforms = [
+      # Could not find a package configuration file provided by "Protobuf"
+      # It is unclear why this is only happening on x86_64-darwin
+      "x86_64-darwin"
+    ];
+  };
+})

